Mathlib/Algebra/Ring/WithAbs.lean

```diff
@@ -97,4 +100,99 @@ theorem equiv_symm_mul :
 /-- `WithAbs.equiv` as a ring equivalence. -/
 def ringEquiv : WithAbs v ≃+* R := RingEquiv.refl _
 
+/-! The completion of a field at an absolute value. -/
+
+variable {K : Type*} [Field K] {v : AbsoluteValue K ℝ}
+  {L : Type*} [NormedField L] {f : WithAbs v →+* L}
+
+/-- If the absolute value `v` factors through an embedding `f` into a normed field, then
+the distance associated to the absolute value also factors through `f`. -/
+theorem dist_of_comp (h : ∀ x, ‖f x‖ = v x) (x y : WithAbs v) : dist x y = dist (f x) (f y) := by
+  rw [(normedField v).dist_eq, (inferInstanceAs <| NormedField L).dist_eq, ← f.map_sub, h]
+  rfl
+
+/-- If the absolute value `v` factors through an embedding `f` into a normed field, then
+the pseudo metric space associated to the absolute value is the same as the pseudo metric space
+induced by `f`. -/
+theorem pseudoMetricSpace_induced_of_comp (h : ∀ x, ‖f x‖ = v x) :
+    (normedField v).toPseudoMetricSpace = PseudoMetricSpace.induced f inferInstance := by
+  ext
+  exact dist_of_comp h _ _
+
+/-- If the absolute value `v` factors through an embedding `f` into a normed field, then
+the uniform structure associated to the absolute value is the same as the uniform structure
+induced by `f`. -/
+theorem uniformSpace_eq_comap_of_comp (h : ∀ x, ‖f x‖ = v x) :
+    (normedField v).toUniformSpace = UniformSpace.comap f inferInstance := by
+  rw [pseudoMetricSpace_induced_of_comp h]
+  rfl
+
+/-- If the absolute value `v` factors through an embedding `f` into a normed field, then
+`f` is uniform inducing. -/
+theorem isUniformInducing_of_comp (h : ∀ x, ‖f x‖ = v x) : IsUniformInducing f :=
+  isUniformInducing_iff_uniformSpace.2 <| Eq.symm (uniformSpace_eq_comap_of_comp h)
+
+/-- If the absolute value `v` factors through an embedding `f` into a normed field, then
+`f` is an isometry. -/
+theorem isometry_of_comp (h : ∀ x, ‖f x‖ = v x) : Isometry f :=
```

Comment:
Ah sorry, I see you already use `Isometry` here. Maybe we can keep only one of the two versions?